apps/src/lab2/views/ExtraLinks.tsx

```diff
@@ -14,61 +16,75 @@ interface PermissionResponse {
   permissions: string[];
 }
 
-interface ExtraLinksResponse {
-  links: {[key: string]: {text: string; url: string; access_key?: string}[]};
-  can_clone: boolean;
-  can_delete: boolean;
-  level_name: string;
-  script_level_path_links: {
-    script: string;
-    path: string;
-  }[];
-}
-
 // If the user has permission to see extra links, fetch extra links for the level,
 // then display a modal with the link data.
 const ExtraLinks: React.FunctionComponent<ExtraLinksProps> = ({
   levelId,
 }: ExtraLinksProps) => {
   const {loading, data} = useFetch('/api/v1/users/current/permissions');
   const [isModalOpen, setIsModalOpen] = useState(false);
-  const [linkData, setLinkData] = useState<ExtraLinksResponse | null>(null);
+  const [levelLinkData, setLevelLinkData] =
+    useState<ExtraLinksLevelData | null>(null);
+  const [projectLinkData, setProjectLinkData] =
+    useState<ExtraLinksProjectData | null>(null);
+  const permissionData = data
+    ? (data as PermissionResponse)
+    : {permissions: []};
+  const channelId = useAppSelector(
+    state => state.lab.channel && state.lab.channel.id
+  );
 
   useEffect(() => {
     const permissionData = data
       ? (data as PermissionResponse)
       : {permissions: []};
-    if (permissionData.permissions.includes(PERMISSIONS.LEVELBUILDER)) {
+    setLevelLinkData(null);
+    setProjectLinkData(null);
+    if (
+      permissionData.permissions.includes(PERMISSIONS.LEVELBUILDER) ||
```

Comment:
nit: can pull these out into locals (`isLevelbuilder` or `hasLevelbuilderPermissions`) since this is checked in multiple if statements?

Reply:
Oh! I was going to do this, but since we fetch permissions in the `useEffect`, I didn't end up assigning a local variable. Thoughts?

Reply:
I think with the refactor you suggested below, this may now be okay as is?